central/scannerdefinitions/handler/handler.go

```diff
@@ -517,8 +418,14 @@ func (h *httpHandler) openMostRecentV4OnlineFile(_ context.Context, t updaterTyp
 	}
 	log.Debugf("Compressed data file is available: %s", openedFile.Name())
 	switch t {
+	case v2UpdaterType:
+		if opts.fileName == "" {
+			return &vulDefFile{File: openedFile, modTime: onlineTime}, nil
+		}
+		fallthrough
 	case mappingUpdaterType:
-		targetFile, cleanUp, err := openFromArchive(openedFile.Name(), opts.mappingFile)
+		defer utils.IgnoreError(openedFile.Close)
+		targetFile, cleanUp, err := openFromArchive(openedFile.Name(), opts.fileName)
```

Comment:
nit: add a note next to this defer to indicate why `openedFile` is closed, something like "`openFromArchive` creates a new file handle, because it does not use the `openedFile` handle we close it to avoid a leak"

Reply:
done here plus two other similar places